v2/introspector.go

```diff
@@ -0,0 +1,68 @@
+package car
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/ipfs/go-cid"
+	carv1 "github.com/ipld/go-car"
+	internalio "github.com/ipld/go-car/v2/internal/io"
+	"io"
+)
+
+// Introspection captures the result of an Introspect call.
+type Introspection struct {
+	// The version, either 1 or 2.
+	Version uint64
+	// The root CIDs if any.
+	Roots []cid.Cid
+	// Indicates whether the index is present.
+	HasIndex bool
+	// The offset at which the CAR v1 starts.
+	// This value will be zero when version is 1.
+	CarV1Offset uint64
+	// The size of the CAR v1 payload.
+	// If the version is 1 this value represents the total number of readable bytes.
+	// If the version is 2 this value is fetched from the v2-specific header.
+	CarV1Size uint64
+	// The offset at which the index starts if it is present, otherwise 0.
+	IndexOffset uint64
+}
+
+// Introspect introspects the given readable bytes and provides metadata about the characteristics
+// and the version of CAR that r represents regardless of its version. This function is backward
+// compatible; it supports both CAR v1 and v2.
+// Returns error if r does not contain a valid CAR payload.
+func Introspect(r io.ReaderAt) (*Introspection, error) {
```

Comment:
This seems like it will be the primary `Read` operation we'll want clients to make, and is the generalized version of `ReadHeader` that previously existed. (where after this generalized header read you can choose to iterate through the items in the car to import them, or open it as a blockstore).

As such, i wonder if we want to call this `Introspect` versus `ReadHeader` or `Parse` or similar?

Reply:
I think `Parse` makes sense; any thoughts on the `Introspection` type name if we call this `Parse`?

`ReadHeader` makes sense in car v1; the issue for v2, however, is that we technically have two headers: the version header (magic prefix that is a valid car v1 header) and the car v2 header. Both these sections have information that a user would want to know as part of introspecting a CAR without knowing its version in advance: 
* version, roots from v1 header, and
* characteristics, offsets from v2 header.

Based on this train of thought I am not sure if `ReadHeader` for introspecting a CAR file that _could be_ v1 or v2 makes sense.

Reply:
ReadHeaders? I agree the name "header" is somewhat overloaded in carv2.

If you prefer to avoid the name "header" altogether, to prevent confusion, I think it's fine to have an API called something else like "inspect" or "introspect".